src/mem/freelist.h

```diff
@@ -121,11 +93,42 @@ namespace snmalloc
     }
 
     /**
-     * Read the next pointer handling any required decoding of the pointer
+     * Static so that it can be used on reference to a FreeObject.
+     *
+     * Returns the next_object field of the next parameter, so can be used to
+     * iterate the construction.
      */
-    CapPtr<FreeObject, CBAlloc> read_next(uint32_t key, LocalEntropy& entropy)
+    static CapPtr<FreeObject, CBAlloc>* store(
```

Comment:
```suggestion
    static CapPtr<FreeObject, CBAlloc>* store_next(
```

Reply:
I agree with this change, but I have another PR to follow this one, which I'll apply this in.